src/utils/logger.h

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright 2021 John Mifsud <imabuddha@gmail.com>
+ *
+ * This file is part of Touchégg.
+ *
+ * Touchégg is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License  as  published by  the  Free Software
+ * Foundation,  either version 3 of the License,  or (at your option)  any later
+ * version.
+ *
+ * Touchégg is distributed in the hope that it will be useful,  but  WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE.  See the  GNU General Public License  for more details.
+ *
+ * You should have received a copy of the  GNU General Public License along with
+ * Touchégg. If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef UTILS_LOGGER_H_
+#define UTILS_LOGGER_H_
+
+#include <string>
+
+enum class LogLevel { INFO, WARNING, ERROR, DEBUG };
+
+/**
+ * USAGE: tlg.LEVEL << "A very informative message!" << std::endl;
+ *
+ * LEVEL: error | warning | info | debug
+ */
+namespace tlg {
+extern LogLevel error;
+extern LogLevel warning;
+extern LogLevel info;
+extern LogLevel debug;
+}  // namespace tlg
+
+/**
+ * Singleton class to provide simple logging functionality.
+ */
+class Logger {
```

Comment:
Do you think it is possible to make Logger private? I don't think it should be part of the interface now that we have tlg::*

Reply:
Not as things currently stand. You see, we need to init the options like this:

```
  // init Logger options
  Logger::obj(args.debug, args.quiet);
```
And _someone_ didn't like the idea of the Logger being coupled to arg parsing so it's being done in main(). 😉

Please feel free to modify that if you like.

Reply:
Yes, I meant something like `tlg::configure(...)` so it is not required to access the Logger class from the outside, but I'll add it.